From 655b1b264635fe49a7a6bcdf8caff83f30766272 Mon Sep 17 00:00:00 2001
From: Ray Strode <rstrode@redhat.com>
Date: Wed, 25 Mar 2015 19:18:45 -0400
Subject: [PATCH 02/10] drm: assume driver doesn't support mapping console

We only support the generic driver now, and it doesn't
support mapping the console.

This commit removes the supports_mapping_console boolean,
and any code that checks for console mapping.
---
 src/plugins/renderers/drm/plugin.c | 31 -------------------------------
 1 file changed, 31 deletions(-)

diff --git a/src/plugins/renderers/drm/plugin.c b/src/plugins/renderers/drm/plugin.c
index 45aab7d..f667b8f 100644
--- a/src/plugins/renderers/drm/plugin.c
+++ b/src/plugins/renderers/drm/plugin.c
@@ -99,7 +99,6 @@ struct _ply_renderer_backend
 
         ply_renderer_driver_interface_t *driver_interface;
         ply_renderer_driver_t           *driver;
-        uint32_t                         driver_supports_mapping_console;
 
         int                              device_fd;
         char                            *device_name;
@@ -456,8 +455,6 @@ load_driver (ply_renderer_backend_t *backend)
         }
 
         backend->driver_interface = ply_renderer_generic_driver_get_interface (device_fd);
-        backend->driver_supports_mapping_console = false;
-
         if (backend->driver_interface == NULL) {
                 close (device_fd);
                 return false;
@@ -732,34 +729,6 @@ create_heads_for_active_connectors (ply_renderer_backend_t *backend)
 
         ply_hashtable_free (heads_by_controller_id);
 
-#ifdef PLY_ENABLE_DEPRECATED_GDM_TRANSITION
-        /* If the driver doesn't support mapping the fb console
-         * then we can't get a smooth crossfade transition to
-         * the display manager unless we use the /dev/fb interface
-         * or the plymouth deactivate interface.
-         *
-         * In multihead configurations, we'd rather have working
-         * multihead, but otherwise bail now.
-         */
-        if (!backend->driver_supports_mapping_console &&
-            ply_list_get_length (backend->heads) == 1) {
-                ply_list_node_t *node;
-                ply_renderer_head_t *head;
-
-                node = ply_list_get_first_node (backend->heads);
-                head = (ply_renderer_head_t *) ply_list_node_get_data (node);
-
-                if (ply_array_get_size (head->connector_ids) == 1) {
-                        ply_trace ("Only one monitor configured, and driver doesn't "
-                                   "support mapping console, so letting frame-buffer "
-                                   "take over");
-
-                        free_heads (backend);
-                        return false;
-                }
-        }
-#endif
-
         return ply_list_get_length (backend->heads) > 0;
 }
 
-- 
1.9.0

